### PR TITLE
[JENKINS-57761] JCasC. A simple test and a complete export/configure roundtrip using new abstract test on JCasC

### PR DESCRIPTION
**Pull request by MRamonLeon**
A complete test exporting and configuring after the restart but using the new abstract class on the JCasC plugin. This way, the plugin only needs to override 3 methods.

I don't add a single test because it's already done with [EssentialsTest](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/integrations/src/test/java/io/jenkins/plugins/casc/EssentialsTest.java)

This PR depends on releasing the new JCasC abstract test: https://github.com/jenkinsci/configuration-as-code-plugin/pull/908 so **we need to update the dependency before merging**

### CODE REVIEW

**Comment by MRamonLeon** [comment]
@oleg-nenashev @batmat @alecharp @fcojfernandez @varyvol The full tests using the generic code moved to JCasC. It will fail until the PR on JCasC is released.

**Comment by MRamonLeon** [comment]
I've changed the dependency to the JCasC plugin to a timestamp deployed to make it green.

**Comment by MRamonLeon** [comment]
Close to reopen to try to run the CI again. Infra failure

File: pom.xml [hunk 1]
```diff
@@ -152,6 +154,19 @@
...
+      <artifactId>configuration-as-code</artifactId>
+      <version>${jcasc.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.jenkins</groupId>
+      <artifactId>configuration-as-code</artifactId>
+      <version>${jcasc.version}</version>
+      <classifier>tests</classifier>
+      <scope>test</scope>
```
Comment (oleg-nenashev) [inline pom.xml]: IMHO we should detach test code to a separate module later. CC @casz @timja

File: src/test/resources/jenkins/metrics/configuration-as-code.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+unclassified:
+  metricsaccesskey:
+    accessKeys:
+      - key:            "tDdG5Vsv-2-WDdHfI3QFPiU9-hcvKmWd2HL4CfVIFvUumQzz3qf6c0qt_HU4_lUh"
```
Comment (oleg-nenashev) [inline src/test/resources/jenkins/metrics/configuration-as-code.yaml]: There were some issues with handling arrays in JCasC. I recommend having a test with takes several items

Comment (MRamonLeon) [reply src/test/resources/jenkins/metrics/configuration-as-code.yaml]: Are you fine if we finish this PR in the current state and I do that in a separate PR? Thanks

File: pom.xml (hunk 1 above)
Comment (MRamonLeon) [reply pom.xml]: Proposed here: https://github.com/jenkinsci/configuration-as-code-plugin/issues/917

File: src/test/resources/jenkins/metrics/configuration-as-code.yaml (hunk 2 above)
Comment (MRamonLeon) [reply src/test/resources/jenkins/metrics/configuration-as-code.yaml]: Or did you mean that https://github.com/jenkinsci/metrics-plugin/pull/46/commits/992748c7502375239ac60fb203a9525ded7c0917?

Comment (oleg-nenashev) [reply src/test/resources/jenkins/metrics/configuration-as-code.yaml]: Yeah, that

**Comment by MRamonLeon** [comment]
@alecharp @dwnusbaum (as Adrien is out next week) Are you ok on merging it?

**Comment by MRamonLeon** [comment]
@dwnusbaum @stephenc Are you ok on merging and releasing it? It could help us to move forward on the CasC effort. Thank you.

**Comment by stephenc** [comment]
Merge and release away. I’m busy elsewhere and it looked fine when last I checked

**Comment by dwnusbaum** [comment]
@MRamonLeon Given I'm not longer working on a team that does anything with this plugin, and that was the only reason I got release permission in the first place, I filed https://github.com/jenkins-infra/repository-permissions-updater/pull/1170 to _replace_ myself with you as a maintainer, so that between you and Adrien you won't be blocked in the future.

**Comment by josejsarmento** [comment]
Would it be possible to add some kind of tutorial to the README to use this? I'd like to configure my jcasc.yaml to have this at startup but I keep getting errors such as:

```
 2022-05-25 10:42:26.487+0000 [id=29]	SEVERE	jenkins.InitReactorRunner$1#onTaskFailed: Failed ConfigurationAsCode.init
 while scanning a simple key
  in /var/jenkins_conf/jenkins.yaml, line 15, column 3:
       metrics
       ^
 could not find expected ':'
  in /var/jenkins_conf/jenkins.yaml, line 16, column 3:
       authorizationStrategy:
       ^
 
 	at org.yaml.snakeyaml.scanner.ScannerImpl.stalePossibleSimpleKeys(ScannerImpl.java:484)
```

**Pull request merged**